apps/web/src/actions/user/setupAction.ts

```diff
@@ -11,38 +11,34 @@ import { z } from 'zod'
 import { errorHandlingProcedure } from '../procedures'
 
 export const setupAction = errorHandlingProcedure
-  .createServerAction()
-  .input(
-    async () => {
-      return z.object({
-        returnTo: z.string().optional(),
-        source: z.string().optional(),
-        name: z.string().min(1, { message: 'Name is a required field' }),
-        email: z
-          .string()
-          .email()
-          .refine(
-            async (email) => {
-              const existingUser = await unsafelyFindUserByEmail(email)
-              return !existingUser
-            },
-            { message: 'Email is already in use' },
-          )
-          .refine(
-            async (email) =>
-              !email.match(/^[A-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[A-Z0-9.-]+$/) &&
-              !email.match(/^[^+]+\+\d+@[A-Z0-9.-]+$/i),
-            { message: 'Email is not valid' },
-          ),
-        companyName: z
-          .string()
-          .min(1, { message: 'Workspace name is a required field' }),
-      })
-    },
-    { type: 'formData' },
+  .inputSchema(
+    z.object({
+      returnTo: z.string().optional(),
+      source: z.string().optional(),
+      name: z.string().min(1, { error: 'Name is a required field' }),
+      email: z
+        .string()
+        .pipe(z.email())
+        .refine(
+          async (email) => {
+            const existingUser = await unsafelyFindUserByEmail(email)
```

Comment:
QA: Check if this async validation continue working

Reply:
it workz